packages/mui-material/src/utils/memoTheme.ts

```diff
@@ -1,31 +1,6 @@
-import { CSSInterpolation } from '@mui/styled-engine';
+import systemMemoTheme from '@mui/system/memoTheme';
 import { Theme } from '../styles/createTheme';
 
-type ThemeStyleFunction = (props: { theme: Theme }) => CSSInterpolation;
+const memoTheme = systemMemoTheme<Theme>;
 
-// We need to pass an argument as `{ theme }` for PigmentCSS, but we don't want to
-// allocate more objects.
-const arg = { theme: undefined as unknown as Theme };
-
-/**
- * Memoize style function on theme.
- * Intended to be used in styled() calls that only need access to the theme.
- */
-export default function memoTheme(styleFn: ThemeStyleFunction) {
-  let lastValue: CSSInterpolation;
-  let lastTheme: Theme;
-
-  return (props: { theme: Theme }) => {
-    let value = lastValue;
-    if (value === undefined || props.theme !== lastTheme) {
-      arg.theme = props.theme;
-
-      value = styleFn(arg);
-
-      lastValue = value;
-      lastTheme = props.theme;
-    }
-
-    return value;
-  };
-}
+export default memoTheme;
```

Comment:
It looks like it shouldn't even be exported #43440

```suggestion
```

Reply:
I was worried about the typings in the initial PR (`memoTheme<T>`), having it here made it easy to include the `MaterialTheme`, though material doesn't use much TS from what I can see. I could remove it altogether from material but it would make it less accessible to external users in case they want it.

Reply:
I want us to move to a point where we force developers to provide their theme, so we need a great DX to not rely on a default theme, so this seems OK.

Reply:
Ok. So should I remove this file and import `unstable_memoTheme` from `@mui/system` everywhere?

Reply:
I think this is beyond the scope of this PR #43440.

Reply:
Btw, I added the export from @mui/material because I wanted to use it in the lab, for the LoadingButton.